Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -794,11 +796,19 @@ extension TaskHandler: ChannelDuplexHandler {
         assert(head.version == HTTPVersion(major: 1, minor: 1),
                "Sending a request in HTTP version \(head.version) which is unsupported by the above `if`")
 
+        self.expectedBodyLength = head.headers[canonicalForm: "content-length"].first.flatMap { Int($0) }
+
         context.write(wrapOutboundOut(.head(head))).map {
             self.callOutToDelegateFireAndForget(value: head, self.delegate.didSendRequestHead)
         }.flatMap {
             self.writeBody(request: request, context: context)
         }.flatMap {
+            if let expectedBodyLength = self.expectedBodyLength, expectedBodyLength != self.actualBodyLength {
+                self.state = .end
```

Comment:
`.end` doesn't sound like a good state for an error, right? We need an `.error` state here I think or else we'll get into trouble later.

Reply:
it might be a good idea, but not for this PR, since it will require to change other places and all switch statements. I don't think that we can get into trouble, since we don't re-use the handler currently, and `.end` is a terminal state anyway, wdyt?

Reply:
@artemredkin is there precedent for using `.end` for error? If yes, I'd rename to `endOrError`

Reply:
yep, there is (since we had no pool before so handler was single-use anyway). `endOrError` sounds great!

Reply:
done